src/python/WMComponent/JobAccountant/AccountantWorker.py

```diff
@@ -493,49 +463,126 @@ def handleSuccessful(self, jobID, fwkJobReport, fwkJobReportPath = None):
                                                 conn = self.getDBConn(),
                                                 transaction = self.existingTransaction())
 
-        fileList = fwkJobReport.getAllFiles()
-        
-        bookKeepingSuccess = True
-        
-        for fwjrFile in fileList:
-            # associate logArchived file for parent jobs on wmstats assuming fileList is length is 1.
-            if jobType == "LogCollect":
+        if jobSuccess:
+            fileList = fwkJobReport.getAllFiles()
+
+            # consistency check comparing outputMap to fileList
+            # they should match except for some limited special cases
+            outputModules = set([])
+            for fwjrFile in fileList:
+                outputModules.add(fwjrFile['outputModule'])
+            if set(outputMap.keys()) == outputModules:
+                pass
+            elif jobType == "LogCollect" and len(outputMap.keys()) == 0 and outputModules == set(['LogCollect']):
```

Comment:
Its a consistency check, I just put in the maximal constraint I can think off. In practical terms just checking for LogCollect job type would be enough too, but as I have the other information available anyways...